test_opensearchpy/run_tests.py

```diff
@@ -94,9 +94,13 @@ def fetch_opensearch_repo() -> None:
         # if the run is interrupted from a previous run, it doesn't clean up, and the git add origin command
         # errors out; this allows the test to continue
         remote_origin_already_exists = 3
-        print(e)
-        if e.returncode != remote_origin_already_exists:
-            sys.exit(1)
+        if e.returncode == remote_origin_already_exists:
+            print(
+                "Consider setting TEST_OPENSEARCH_NOFETCH=true if you want to reuse the existing local OpenSearch repo"
+            )
+        else:
+            print(e)
+        sys.exit(1)
```

Comment:
should this `sys.exit(1)` be moved into else?

Reply:
My intent here was to make it clear why the exit was happening. I didn't want to change the previous behavior especially after I found the TEST_OPENSEARCH_NOFETCH variable.

Reply:
@macohen, I apologize if I am causing any confusion, but I believe there's a change in the behavior for `e.returncode == remote_origin_already_exists`.

It seems that there shouldn't be a `sys.exit(1)` when `e.returncode == remote_origin_already_exists`. Please correct me if I'm mistaken.

Reply:
Sorry for the poor explanation on my end. Prior to my change, when the git repository already had a remote origin added, the whole process exited. I was considering just ignoring that state, but then found the TEST_OPENSEARCH_NOFETCH variable which can be set when you want that state to be ignored. I figured it would be better to maintain the behavior of the script in that case and also tell the user that TEST_OPENSEARCH_NOFETCH could help if exiting was not desired. I hope that helps...